src/cljc/rems/common/util.cljc

```diff
@@ -19,6 +19,100 @@
   (is (= nil
          (re-matches +phone-number-regex+ "+35845000000000000000000000000000"))))
 
+;; regex from https://stackoverflow.com/questions/5284147/validating-ipv4-addresses-with-regexp
+(def +valid-ip-address-regex+ #"(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$")
+
+(def +valid-ip-address-regex-version-six+ #"(([0-9]|[a-f]){1,4}(\:)){7}(([0-9]|[a-f]){1,4})$")
+
+;; https://stackoverflow.com/questions/2814002/private-ip-address-identifier-in-regular-expression
+;; for now, this test should test for the following private IPv4 address pattaerns
+;; 0.x.x.x
+;; 10.x.x.x - Private network, local communications
+;; 100.64.x.x–100.127.x.x - Private network, communications between a service provider and its subscribers
+;; 127.x.x.x - host, loopback addresses to the local host
+;; 169.254.x.x - Subnet, Used for link-local addresses[7] between two hosts on a single link when no IP address is otherwise specified
+;; 172.16.x.x – 172.31.x.x - Private network, local communications within a private network
+;; 192.0.0.x - Private network, IETF Protocol Assignments
+;; 192.0.2.x - Documentation, Assigned as TEST-NET-1, documentation and examples.
+;; 192.168.x.x - Private network, local communications within a private network
+;; 198.18.x.x - 198.19.x.x - Private network, benchmark testing of inter-network communications between two separate subnets
+;; 198.51.100.x - Documentation, Assigned as TEST-NET-2, documentation and examples.
+;; 203.0.113.x - Documentation, Assigned as TEST-NET-3, documentation and examples.
+;; 240.x.x.x–255.x.x.x - multicast
+(def +reserved-ip-address-range-regex+
+  (re-pattern
+   (str
+    "((0|10|127|224?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "((100?)\\.([6-9][4-9]|1[1-2][1-7]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "((169?)\\.(254?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
```

Comment:
This isn't correct either: the extra `?` characters cause ip addresses like 16.254.x.x and 169.25.x.x to be marked as reserved. Test case:

```
  (is (= nil
         (first (re-matches +reserved-ip-address-range-regex+ "16.254.1.1"))))
  (is (= nil
         (first (re-matches +reserved-ip-address-range-regex+ "169.25.1.1"))))
```